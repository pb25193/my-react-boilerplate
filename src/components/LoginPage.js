import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Boilerplate</h1>
            <p>Tagline for your app!</p>
            <button
                className="button"
                onClick={startLogin}
            >Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch)=>({
    startLogin: () => dispatch(startLogin())
});

const ConnectedLoginPage = connect(undefined, mapDispatchToProps)(LoginPage);

export { LoginPage, ConnectedLoginPage as default };