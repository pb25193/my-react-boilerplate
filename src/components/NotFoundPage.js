import React from 'react';



const NotFoundPage = () => (
    <div>
        Sorry! It appears that the page you are looking for does not exist.
    </div>
);


export default NotFoundPage;