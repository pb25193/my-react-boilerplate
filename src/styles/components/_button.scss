.button {
    border: none;
    color: white;
    padding: $s-size;
    font-weight: 300;
    font-size: $font-size-large;
    background: $blue;
    line-height: 1;
    display: inline-block;
    text-decoration: none;
}

.button--danger {
    background: $red;
}


.button--link {
    background: none;
}