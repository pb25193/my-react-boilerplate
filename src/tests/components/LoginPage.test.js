import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test("should show up", ()=>{
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('should log in correctly', ()=>{
    const logInBtn = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={logInBtn} />);
    wrapper.find('button').simulate('click');
    expect(logInBtn).toHaveBeenCalled();
})