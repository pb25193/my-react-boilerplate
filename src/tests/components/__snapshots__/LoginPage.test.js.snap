// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show up 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerplate
    </h1>
    <p>
      Tagline for your app!
    </p>
    <button
      className="button"
    >
      Login with Google
    </button>
  </div>
</div>
`;
